Source/WebCore/rendering/line/BreakingContext.h

```diff
@@ -364,7 +364,26 @@ inline void BreakingContext::handleOutOfFlowPositioned(Vector<RenderBox*>& posit
 inline void BreakingContext::handleFloat()
 {
     auto& floatBox = downcast<RenderBox>(*m_current.renderer());
-    const auto& floatingObject = *m_lineBreaker.insertFloatingObject(floatBox);
+    auto& floatingObject = *m_lineBreaker.insertFloatingObject(floatBox);
+
+    auto wouldFitWithTrimmedMargin = [&](MarginTrimType marginTrimType) {
+        auto widthWitoutMargin = floatingObject.width();
+        widthWitoutMargin -= marginTrimType == MarginTrimType::InlineStart ? floatBox.marginStart(&m_blockStyle) : floatBox.marginEnd(&m_blockStyle);
+        return m_width.fitsOnLineExcludingTrailingWhitespace(widthWitoutMargin.toFloat());
```

Comment:
please check if it works with clear; floats with clear have a very different horizontal space requirements.